test/prevent-abbreviations.js

```diff
@@ -0,0 +1,400 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/prevent-abbreviations';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	},
+	parserOptions: {
+		sourceType: 'module'
+	}
+});
+
+const noFixingTestCase = test => Object.assign({}, test, {
+	output: test.code
+});
+
+const createErrors = message => {
+	const error = {
+		ruleId: 'prevent-abbreviations'
+	};
+
+	if (message) {
+		error.message = message;
+	}
+
+	return [error];
+};
+
+const extendedOptions = [{
+	replacements: {
+		e: false,
+		c: {
+			custom: true
+		},
+		cb: {
+			callback: false,
+			circuitBreacker: true
+		}
+	}
+}];
+
+const customOptions = [{
+	extendDefaults: false,
+	replacements: {
+		args: {
+			arguments: true
+		},
+		e: {
+			error: true,
+			event: true,
+			element: true
+		},
+		err: {
+			error: true
+		}
+	}
+}];
+
+ruleTester.run('prevent-abbreviations', rule, {
+	valid: [
+		'let x',
+		'let error',
+		'error => {}',
+		'(function (error) {})',
+		'function f(x) {}',
+
+		// `err` in not defined, should not be report (could be reported by `no-unused-vars`)
+		'console.log(err)',
+
+		'let c',
+		{
+			code: 'let c',
+			options: customOptions
+		},
+
+		{
+			code: 'let cb',
+			options: customOptions
+		},
+
+		{
+			code: 'let e',
+			options: extendedOptions
+		},
+
+		// Renaming to `arguments` would result in a `SyntaxError`, so it should keep `args`
+		`
+			class A {
+				method(...args) {
+					return super.method(...args) + 1;
+				}
+			}
+		`,
+		`
+			class A extends B {
+				constructor(...args) {
+					super(...args);
+				}
+			}
+		`,
+
+		// Renaming to `arguments` whould result in `f` returning it's arguments instead of the outer variable
+		`
+			let args;
+			function f() {
+				return args;
+			}
+		`,
+
+		`
+			let args;
+			function f() {
+				return arguments + args;
+			}
+		`,
+
+		`
+			let args;
+			function f() {
+				return g.apply(this, arguments) + args;
+			}
+		`,
+
+		// Renaming would change the API of the module
+		'export const err = {}',
+		'export let err = {}',
+		'export var err = {}',
+		'export function err() {}',
+		'export class err {}'
```

Comment:
@sindresorhus I'm not sure what to do with these cases.

* Fix to this:
```js
function error() {}
export {error as err}
```
* Not report
* Report but not fix

What do you think?

Reply:
> * Report but not fix

👍